From dc7ec5740b346fa93988b656927c8ec4eca5b155 Mon Sep 17 00:00:00 2001
From: "Manu.WTF" <43337106+Dark4MD@users.noreply.github.com>
Date: Wed, 16 Jan 2019 14:43:16 +0100
Subject: [PATCH] ramips-mt7620: add support for Nexx WT3020 (8M)

---
 docs/index.rst        | 4 ++++
 targets/ramips-mt7620 | 8 ++++++++
 2 files changed, 12 insertions(+)

diff --git a/docs/index.rst b/docs/index.rst
index 40b7af16..69b2cd1f 100644
--- a/docs/index.rst
+++ b/docs/index.rst
@@ -352,6 +352,10 @@ ramips-mt7620
   - GL-MT300N [#80211s]_
   - GL-MT750 [#80211s]_
 
+* Nexx
+
+  - WT3020AD/F/H
+
 ramips-mt7621
 ^^^^^^^^^^^^^
 
diff --git a/targets/ramips-mt7620 b/targets/ramips-mt7620
index 2f70cd65..e0b15196 100644
--- a/targets/ramips-mt7620
+++ b/targets/ramips-mt7620
@@ -1,4 +1,5 @@
 # GL Innovations
+
 device gl-mt300a gl-mt300a
 factory
 
@@ -7,3 +8,10 @@ factory
 
 device gl-mt750 gl-mt750
 factory
+
+# Nexx
+
+device nexx-wt3020-8m wt3020-8M
+alias nexx-wt3020ad
+alias nexx-wt3020f
+alias nexx-wt3020h
-- 
2.17.1

