From bb6338af36834f828bf44de8149a176fccfd08ac Mon Sep 17 00:00:00 2001
From: Martin Weinelt <martin@darmstadt.freifunk.net>
Date: Sun, 3 Feb 2019 21:15:32 +0100
Subject: [PATCH] brcm2708-bcm2708: add manifest alias for Raspberry Pi Model B
 Plus Rev 1.2

---
 targets/brcm2708-bcm2708 | 1 +
 1 file changed, 1 insertion(+)

diff --git a/targets/brcm2708-bcm2708 b/targets/brcm2708-bcm2708
index c0d8b8ea..1fe6f390 100644
--- a/targets/brcm2708-bcm2708
+++ b/targets/brcm2708-bcm2708
@@ -2,3 +2,4 @@
 
 device raspberry-pi rpi
 manifest_alias raspberry-pi-model-b-rev-2
+manifest_alias raspberry-pi-model-b-plus-rev-1.2
-- 
2.17.1

